model/private_subnet.rb

```diff
@@ -53,7 +59,11 @@ def display_state
   semaphore :destroy, :refresh_keys, :add_new_nic, :update_firewall_rules
 
   def self.random_subnet
-    PRIVATE_SUBNET_RANGES.sample
+    subnet_dict = PRIVATE_SUBNET_RANGES.each_with_object({}) do |subnet, hash|
+      prefix_length = subnet.split("/").last.to_i
```

Comment:
Consider `Integer(s, 10)` because `.to_i` is always one edit away from "0" given erroneous input. the `10` is necessary to avoid, say, octal conversion.